reddit_decider/__init__.py

```diff
@@ -269,6 +269,89 @@ def get_variant_without_expose(self, experiment_name: str) -> Optional[str]:
     #     self, experiment_name: str, variant_name: str, **exposure_kwargs: Optional[Dict[str, Any]]
     # ) -> None:
 
+    def get_variant_for_identifier(
+        self,
+        experiment_name: str,
+        identifier: str,
+        **exposure_kwargs: Optional[Dict[str, Any]]
+    ) -> Optional[str]:
+        """Return a bucketing variant for `identifier`, if any, with auto-exposure.
+
+        The `identifier` param will be set on `DeciderClient` under:
+            `user_id`, `device_id`, & `canonical_url`,
+        so that regardless of what `bucketing_val` is set to in an experiment's config
+        (one of those 3), the passed in `identifier` param will be used in the
+        hashing string when bucketing.
+
+        Since calling `get_variant_for_identifier()` will fire an exposure event, it
+        is best to call it when you are sure the user will be exposed to the experiment.
```

Comment:
We have this for `get_variant()`: 
```
If you absolutely must check the status of an experiment before the user will be exposed to the experiment,
use `get_variant_without_expose()` to disable exposure events and call `expose()` manually later.
```
Should we also have a `get_variant_for_identifier_without_expose()` to match functionality?

Reply:
you are right. We should probably have a `get_variant_for_identifier_without_expose()` as well

Reply:
will create follow-up pr for that.